src/Serializer.js

```diff
@@ -40,7 +40,7 @@ class Serializer {
         }
 
         // drop FateType prefix
-        const typeName = data.constructor.name.slice(4)
+        const typeName = data.typeName || data.constructor.name.slice(4)
```

Comment:
having two classes with the same class name is a tricky thing, proposing this as a workaround